### DISCUSSION TITLE
communication problems

### DISCUSSION THREAD

**Discussion by Matze5593** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#13] [Q&A]
Hello, I flashed and connected the ESP32 D1 mini according to the instructions. The whole thing worked so far. I just don't get any data

how can I find the cause?

![Screenshot 2024-01-18 173722](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/3c8ba98d-15f9-4299-8bdc-057ee6ef9462)
![20240118_170038](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/83b6d063-fcfc-45e4-a790-b5f2c8daf7dd)
![20240118_170220](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/e1952d28-652a-4084-af29-2aa88cd83d6d)
![20240118_170234](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/39bad923-a24d-46c2-889e-97559ab2574d)
![20240118_170329](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/04ed0687-a804-4370-a101-14ea86d92dad)
![20240118_170338](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/603700b6-5bde-492e-a2ec-1413d3f9b16f)

**Comment by wavoigt** [comment]
1. You use the VCC Pin (5V) to supply the RS232 converter, you should use 3V3 from D1 mini board, because of the correct logic levels.
2. You use Pins tx/rx on the D1 mini board, this is hardwired to the onboard USB-RS232 converter.
    You should use pins 16/17:

Froeling.yaml (part):

<pre>esp32:
  board: wemos_d1_mini32
  framework:
  type: arduino
uart:
  #alternative UART2 tx_pin: (1) 17 rx_pin: (3) 16
  id: uartfroeling
  tx_pin: 17
  rx_pin: 16
  baud_rate: 57600
  data_bits: 8
  stop_bits: 1
  parity: NONE
</pre>

What do you see when connecting by browser to D1 mini via WiFi?
Is there a list of entities and a log output?
Did you try to swap rx / tx wires, because some RS232 adapters might be not named correctly?

see also:
https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/issues/6

**Reply by Matze5593** [reply]
Ok, but in the readme it is recorded exactly as I connected it.

![Screenshot_20240118_180820_GitHub.jpg](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/a5e5da64-4b34-4893-b759-0dfef1964d2e)

**Comment by wavoigt** [comment]
May be, it was only a suggestion.
With my configuration, all is working fine.
You'll find more informations in the closed issues...

**Comment by Matze5593** [comment]
Good job, she was right. Swapping RX & TX was the solution THX

I used these components.  Manufacturer mislabeled


![Screenshot_20240118_182003_Amazon Shopping.jpg](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/55891473/ff91f84e-3d23-4013-b038-b1adf50db271)

https://amzn.eu/d/bV4CEs6
